### PR TITLE
Rename Advanced Throwing component to be in-line with other advanced_ components

### PR DESCRIPTION
**Pull request by jonpas**
**When merged this pull request will:**
- Rename `advancedthrowing` to `advanced_throwing`, to mimic the already existent `advanced_` components
- Update documentation with new name

Because @BaerMitUmlaut was nagging me about it. I have no idea what to tag this with.

### CODE REVIEW

File: docs/wiki/feature/advanced-throwing.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ version:
   major: 3
   minor: 7
   patch: 0
+redirect_from: "/wiki/feature/advancedthrowing.html"
```
Comment (bux578) [inline docs/wiki/feature/advanced-throwing.md]: :smile:

Comment (jonpas) [inline docs/wiki/feature/advanced-throwing.md]: I can remove that if you think it's not necessary. :P

Comment (bux578) [inline docs/wiki/feature/advanced-throwing.md]: I just skimmed through this PR to see if I would find that line :)

Comment (jonpas) [inline docs/wiki/feature/advanced-throwing.md]: Haha, do I get points for it? :laughing:

**Comment by jonpas** [comment]
For reference, @BaerMitUmlaut was being super anal about it, I just refused to let him touch my code.

**Pull request merged**